src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -154,7 +154,10 @@ public void Dispose()
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                _solutionEvents.AfterClosing -= SolutionEvents_AfterClosing;
+                if (_solutionEvents != null)
```

Comment:
On shutting down VisualStudio with the debugger attached I got a nullref exception there, which is why I added the null check.